modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,906 @@
+    /** @brief This function partitions samples according to the split
+    * @param split stores the split according to which the samples would  be split
+    * @param samples vector containing all the training samples
+    * @param start integer storing the start index of the training samples
+    * @param end integer storing the end index of training samples
+    */
+    unsigned long divideSamples (splitr split,vector<training_sample>& samples,unsigned long start,unsigned long end);
+    /** @brief This function fits a regression tree according to the shape residuals calculated to give weak learners for GBT algorithm.
+    *@param tree stores an empty regression tree.
+    *@param samples vector storing the training samples for building the regression tree.
+    * @param pixel_coordinates vector storing the test pixel coordinates from which the coordinates of pixels for making split are selected.
+    */
+    bool buildRegtree(regtree &tree,vector<training_sample>& samples,vector<Point2f> pixel_coordinates);
+    /** @brief This function greedily decides the best split among the test splits generated.
+    * @param samples vector storing the training samples
+    * @param pixel_coordinates vector for storing the random test coordinates generated..
+    * @param start stores start index of the samples currently at the node
+    * @param end stores end index of the samples currently at the node
+    */
+    splitr generateSplit(vector<Point2f> pixel_coordinates, vector<training_sample> samples,unsigned long start ,unsigned long end);
+    /**@brief This function randomly generates test coordinates for each level of cascade.
+    * @param pixel_coordinates stores the test coordinates generated for making splits.
+    * @param min_x minimum x coordinate of the random pixels generated
+    * @param max_x maximum x coordinate of the random pixels generated
+    * @param min_y minimum y coordinate of the random pixels generated
+    * @param max_y maximum y coordinate of the random pixels generated
+    */
+    void getTestCoordinates (vector< vector<Point2f> >& pixel_coordinates,float min_x,
+                                        float min_y, float max_x , float max_y);
+    /** @brief This function gets the landmarks in the meanshape nearest to the pixel coordinates.
+    * @param pixels stores the point nearest to which shape coorinates are to be determined.
+    */
+    unsigned long getNearestLandmark (Point2f pixels );
+    /** @brief This function implements gradient boosting by fitting regression trees
+    * to get weak learners and update the shape residual.
+    *@returns A forest of regression trees fitted which act as weak learners for the model.
+    *@param samples stores the training samples.
+    *@param pixel_coordinates stores the test pixel coordinates for making the split.
+    */
+    vector<regtree> gradientBoosting(vector<training_sample>& samples,vector<Point2f> pixel_coordinates);
```

Comment:
For everything above this
- Function documentation is good but needs to move to header file
- Functions not publicly accessible can have a small documentation in source, but do not need full doxygen style
- Also if a function should get a documentation as expanded as this, to me it seems this should be in the public interface and thus in the header file, at least the function declaration.